cms/lib/xblock/tagging/tagging.py

```diff
@@ -47,18 +47,21 @@ def student_view_aside(self, block, context):  # pylint: disable=unused-argument
             tags = []
             for tag in self.get_available_tags():
                 values = tag.get_values()
-                current_value = self.saved_tags.get(tag.name, None)
+                current_values = self.saved_tags.get(tag.name, [])
 
-                if current_value is not None and current_value not in values:
-                    values.insert(0, current_value)
+                if isinstance(current_values, basestring):
+                    current_values = [current_values]
+
+                values_not_exists = [cur_val for cur_val in current_values if cur_val not in values]
 
                 tags.append({
                     'key': tag.name,
                     'title': tag.title,
                     'values': values,
-                    'current_value': current_value
+                    'current_values': values_not_exists + current_values,
```

Comment:
Fixed this. There was not big issue with duplicates because `current_values` were used only to highlight the selected values in the selector, but of course it is unpredictable behavior so I've fixed it. Thanks.